Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts

```diff
@@ -0,0 +1,156 @@
+                }
+                tl.setResult(tl.TaskResult.Failed, tl.loc('ServerCallFailed'));
+            }
+        })
+        .auth(cred.mUsername, cred.mPassword, true)
+        .on('error', err => {
+            tl.setResult(tl.TaskResult.Failed, err);
+        });
+}
+
+function getLastSuccessful(serverEndpointUrl: string, jobName: string, cred: Credential, strictSSL: boolean): Q.Promise<number> {
+    let defer = Q.defer<number>();
+    let lastSuccessfulUrl = `${serverEndpointUrl}/job/${jobName}/api/json?tree=lastSuccessfulBuild[id,displayname]`;
+
+    getRequest(lastSuccessfulUrl, cred, strictSSL, result => {
+        if (result && result['lastSuccessfulBuild']) {
+            let lastSuccessfulBuildId = result['lastSuccessfulBuild']['id'];
+            if (lastSuccessfulBuildId) {
+                defer.resolve(lastSuccessfulBuildId);
+            } else {
+                tl.setResult(tl.TaskResult.Failed, tl.loc('CouldNotGetLastSuccessfuilBuildNumber'));
+            }
+        } else {
+            tl.setResult(tl.TaskResult.Failed, tl.loc('CouldNotGetLastSuccessfuilBuildNumber'));
+        }
+    });
+              
+    return defer.promise;
+}
+
+function getArtifactsRelativePaths(serverEndpointUrl: string, jobName: string, jobBuildId: number, 
+        cred: Credential, strictSSL: boolean): Q.Promise<string[]> {
+    let defer = Q.defer<string[]>();
+    let artifactQueryUrl = `${serverEndpointUrl}/job/${jobName}/${jobBuildId}/api/json?tree=artifacts[*]`;
+
+    getRequest(artifactQueryUrl, cred, strictSSL, result => {
+        if (result && result['artifacts']) {
+            let artifacts = result['artifacts'];
+            if (artifacts.length === 0) {
+                tl.setResult(tl.TaskResult.Failed, tl.loc('CouldNotFindArtifacts', jobName, jobBuildId));
+            } else {
+                let articactsRelativePath = result['artifacts'].map(artifact => {
+                    return artifact['relativePath'];
+                });
+                defer.resolve(articactsRelativePath);
+            }
+        } else {
+            // no artifacts for this job
+            tl.setResult(tl.TaskResult.Failed, tl.loc('CouldNotFindArtifacts', jobName, jobBuildId));
+        }
+    });
+
+    return defer.promise 
+}
+
+async function download(url: string, localFile: string, cred: Credential, strictSSL: boolean) {
+    tl.debug(tl.loc("DownloadFileTo", url, localFile));
+    await request.get( {url: url, strictSSL: strictSSL})
+        .auth(cred.mUsername, cred.mPassword, true)
+        .on('error', err => {
+            tl.setResult(tl.TaskResult.Failed, err)
+        })
+        .pipe(fs.createWriteStream(localFile));
+
+    tl.debug(tl.loc("FileSuccessfullyDownloaded", localFile));
+}
+
+function getArtifactUrl(serverEndpointUrl: string, jobName: string, jobBuildId: number, relativePath: string) {
+    return `${serverEndpointUrl}/job/${jobName}/${jobBuildId}/artifact/${relativePath}`;
```

Comment:
To prevent manipulation of this URL, we could fail if `jobName` includes slashes.